lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -78,8 +78,11 @@ class ThreadRepositoryImpl extends ThreadRepository {
       yield localEmailResponse;
     }
 
-    if (localEmailResponse.hasState() && networkEmailResponse == null) {
+    if (localEmailResponse.hasState()) {
       log('ThreadRepositoryImpl::getAllEmail(): filter = ${emailFilter?.mailboxId} local has state: ${localEmailResponse.state}');
+      if (networkEmailResponse != null) {
+        await _updateEmailCache(newCreated: networkEmailResponse.emailList);
```

Comment:
- As I understand it, we will always update the email cache again when `networkReponse` is different from `null`. And should we update `state` again?.
- Also you should execute that statement before checking the condition `if/else`


<img width="854" alt="Screen Shot 2022-10-05 at 09 19 49" src="https://user-images.githubusercontent.com/80730648/193966035-79af1542-fe0f-45ce-9002-c7ce3a7c4066.png">

Reply:
fixed